Resources/public/js/views/fields/ez-relation-editview.js

```diff
@@ -0,0 +1,195 @@
+                    that._set('destinationContentId', that.get('destinationContent').get('contentId'));
+                } else {
+                    that._set('destinationContentId', null);
+                }
+            });
+        },
+
+        /**
+         * Fire the `loadFieldRelatedContent` event
+         *
+         * @method _fireLoadFieldRelatedContent
+         * @protected
+         */
+        _fireLoadFieldRelatedContent: function () {
+            if ( !this._isFieldEmpty() ) {
+                this.fire('loadFieldRelatedContent', {
+                    fieldDefinitionIdentifier: this.get('fieldDefinition').identifier
+                });
+            }
+        },
+
+        /**
+         * Checks whether the field is empty
+         *
+         * @method _isFieldEmpty
+         * @protected
+         * @return {Boolean}
+         */
+        _isFieldEmpty: function () {
+            return !this.get('destinationContentId');
+        },
+
+        /**
+         * Returns an object containing the additional variables
+         *
+         * @method _variables
+         * @protected
+         * @return Object
+         */
+        _variables: function () {
+            var dest = this.get('destinationContent');
+
+            return {
+                destinationContent: dest ? dest.toJSON() : null,
+                loadingError: this.get('loadingError'),
+                isEmpty: this._isFieldEmpty(),
+                isRequired: this.get('fieldDefinition').isRequired,
+            };
+        },
+
+        /**
+         * Tap event handler for the remove button. It resets the relation.
+         *
+         * @method _removeRelation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _removeRelation: function (e) {
+            e.preventDefault();
+
+            this.set('destinationContent', null);
+            this.validate();
+            this.get('container').one('.ez-relationeditview-remove').set('disabled', true);
+        },
+
+        validate: function () {
+            if ( this.get('fieldDefinition').isRequired && this._isFieldEmpty() ){
+                this.set('errorStatus', 'This field is required');
+            } else {
+                this.set('errorStatus', false);
+            }
+        },
+
+        /**
+         * Fire the contentDiscover event with the universalDiscovery.
+         *
+         * @method _runUniversalDiscovery
+         * @protected
+         * @param {Object} e
+         */
+        _runUniversalDiscovery: function (e) {
+            e.preventDefault();
+            this.fire('contentDiscover', {
+                config: {
+                    contentDiscoveredHandler: Y.bind(this._selectRelation, this),
+                    cancelDiscoverHandler: Y.bind(this.validate, this),
+                },
+            });
+        },
+
+        /**
+         * Select the relation.
+         * Set destinationContent and destinationContentId attributes.
```

Comment:
wrong now. But the whole doc should be replaced by something like: *Universal discovery contentDiscovered event handler to fill the relation after the user chose a content*